sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -64,6 +65,19 @@ class SparkSession private(
    |  Session-related state  |
    * ----------------------- */
 
+  {
+    val defaultWarehousePath =
+      SQLConf.WAREHOUSE_PATH
+        .defaultValueString
+        .replace("${system:user.dir}", System.getProperty("user.dir"))
+    val warehousePath = sparkContext.conf.get(
+      SQLConf.WAREHOUSE_PATH.key,
+      defaultWarehousePath)
+    sparkContext.conf.set(SQLConf.WAREHOUSE_PATH.key, warehousePath)
+    sparkContext.conf.set("hive.metastore.warehouse.dir", warehousePath)
```

Comment:
At runtime, if users change the value of `SQLConf.WAREHOUSE_PATH.key` by using the `Set` Command, we still need to set `hive.metastore.warehouse.dir`. Right? 

In addition, I think we should disallow users to change the value of `hive.metastore.warehouse.dir` by using the `Set` Command.